### PR TITLE
gpg2: inline stdin write

### PR DESCRIPTION
**Pull request by jvehent**

### CODE REVIEW

File: signer/gpg2/gpg2.go [hunk 1]
```diff
@@ -193,6 +199,10 @@ func (s *GPG2Signer) SignData(data []byte, options interface{}) (signer.Signatur
 	defer os.Remove(tmpContentFile.Name())
 	ioutil.WriteFile(tmpContentFile.Name(), data, 0755)
 
+	// take a mutex to prevent multiple invocations of gpg in parallel
+	serializeSigning.Lock()
+	defer serializeSigning.Unlock()
```
Comment (g-k) [inline signer/gpg2/gpg2.go]: is this lock freed if there's a pipe write or other error?

**Comment by g-k** [comment]
Also, we should make an issue to unit or integration test the failure case.

**Comment by jvehent** [comment]
> Also, we should make an issue to unit or integration test the failure case.

I'm not even sure how we could reproduce it reliably. Perhaps this is similar to https://github.com/mozilla-services/autograph/blob/master/database/queries_test.go#L11 ?

**Comment by g-k** [comment]
Yeah might not be worth the effort.

**Pull request merged**